bus-mapping/src/evm/opcodes/begin_end_tx.rs

```diff
@@ -615,3 +618,57 @@ fn gen_tx_l1_fee_ops(
     )?;
     Ok(())
 }
+
+// Add two copy-events for tx access-list addresses and storage keys if EIP-2930.
+fn gen_tx_eip2930_ops(
+    state: &mut CircuitInputStateRef,
+    exec_step: &mut ExecStep,
+) -> Result<(), Error> {
+    if !state.tx.tx_type.is_eip2930_tx() {
+        return Ok(());
+    }
+
+    let tx_id = NumberOrHash::Number(state.tx_ctx.id());
+    let (address_size, storage_key_size) =
+        access_list_address_and_storage_key_sizes(&state.tx.access_list);
+
+    // Add copy event for access-list addresses.
+    let rw_counter_start = state.block_ctx.rwc;
+    state.push_copy(
+        exec_step,
+        CopyEvent {
+            src_addr: 0,
+            src_addr_end: address_size,
+            src_type: CopyDataType::TxAccessListAddresses,
+            src_id: tx_id.clone(),
+            dst_addr: 0,
+            dst_type: CopyDataType::RwAccessListAddresses,
+            dst_id: tx_id.clone(),
+            log_id: None,
+            rw_counter_start,
+            // TODO
+            copy_bytes: CopyBytes::new(vec![], None, None),
+        },
+    );
+
+    // Add copy event for access-list storage keys.
+    let rw_counter_start = state.block_ctx.rwc;
+    state.push_copy(
+        exec_step,
+        CopyEvent {
+            src_addr: 0,
+            src_addr_end: storage_key_size,
+            src_type: CopyDataType::TxAccessListStorageKeys,
+            src_id: tx_id.clone(),
```

Comment:
you may not need this clone here. Do you?

Reply:
Seems it's needed, since it's constructed as [NumberOrHash::Number here](https://github.com/scroll-tech/zkevm-circuits/pull/1045/files/10436947f181c545f5f3950f49be0b1386ab6d8a#diff-d8df2606a9422d81101a23b7e700280608c770cc3c98b6f214dd8e207835efc3R631) and also used as [dst_id here](https://github.com/scroll-tech/zkevm-circuits/pull/1045/files/10436947f181c545f5f3950f49be0b1386ab6d8a#diff-d8df2606a9422d81101a23b7e700280608c770cc3c98b6f214dd8e207835efc3R665).